scipy/optimize/optimize.py

```diff
@@ -2102,7 +2105,10 @@ def _minimize_scalar_golden(func, brack=None, args=(),
     f1 = func(*((x1,) + args))
     f2 = func(*((x2,) + args))
     funcalls += 2
-    while (numpy.abs(x3 - x0) > tol * (numpy.abs(x1) + numpy.abs(x2))):
+    nit = 0
+    for i in xrange(maxiter):
```

Comment:
In that case we count iteration before it starts, but in the loop we have is statement that checks whether we can run one more iteration or not. Which means that we cannot increment ``nit`` variable until we know that loop hasn't been terminated. 

```python
>>> nit = 0
>>> maxiter = 1
>>> for nit in xrange(1, maxiter + 1):
...     if False:
...         break
...
>>> nit
1
```